corehq/motech/repeaters/tasks.py

```diff
@@ -554,6 +556,18 @@ def _get_lock(self, repeater_id):
         return lock
 
 
+@periodic_task(
+    run_every=crontab(hour='5', minute='0', day_of_month='1'),
+    queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'),
+)
+def purge_old_datasourceupdates():
+    table_name_format = f'{DataSourceUpdate.Meta.db_table}_y%Ym%m'
+    max_age = datetime.utcnow() - DataSourceUpdate.MAX_AGE
+    table_name = (max_age - timedelta(days=1)).strftime(table_name_format)
+    with connection.cursor() as cursor:
+        cursor.execute(f'DROP TABLE IF EXISTS {table_name}')
```

Comment:
Not a huge deal, but this only looks for the table from 4 months ago right? If for some reason this task failed for a period of time, we might end up with tables that don't get cleaned up unless we manually intervene. This has happened with synclogs before, and [the cleanup logic](https://github.com/dimagi/commcare-hq/blob/e038adaf12996f3de193d93a9732df79313ef593/corehq/ex-submodules/casexml/apps/phone/tasks.py#L81-L91) used there is a bit more thorough if you want to adapt it to here as well.

Reply:
Interesting! 35d046c290e

We should do the same for [device report entries](corehq/ex-submodules/phonelog/tasks.py:18).